src/query/db_accessor.hpp

```diff
@@ -406,6 +406,47 @@ class DbAccessor final {
     return std::make_optional<VertexAccessor>(*value);
   }
 
+  storage::Result<std::optional<std::pair<std::vector<VertexAccessor>, std::vector<EdgeAccessor>>>>
+  DetachDeleteVertexBulk(std::vector<VertexAccessor> nodes, std::vector<EdgeAccessor> edges, bool detach) {
+    using ReturnType = std::pair<std::vector<VertexAccessor>, std::vector<EdgeAccessor>>;
+
+    std::vector<storage::VertexAccessor> nodes_impl;
+    std::vector<storage::EdgeAccessor> edges_impl;
+
+    nodes_impl.reserve(nodes.size());
+    edges_impl.reserve(edges.size());
+
+    std::transform(nodes.begin(), nodes.end(), std::back_inserter(nodes_impl),
+                   [](const auto &node_info) { return node_info.impl_; });
+    std::transform(edges.begin(), edges.end(), std::back_inserter(edges_impl),
+                   [](const auto &edge_info) { return edge_info.impl_; });
+
+    auto res = accessor_->DetachDeleteVertexBulk(std::move(nodes_impl), std::move(edges_impl), detach);
+    if (res.HasError()) {
+      return res.GetError();
+    }
+
+    const auto &value = res.GetValue();
+    if (!value) {
+      return std::optional<ReturnType>{};
+    }
+
+    const auto &[val_vertices, val_edges] = *value;
+
+    std::vector<VertexAccessor> deleted_vertices;
+    std::vector<EdgeAccessor> deleted_edges;
+
+    deleted_vertices.reserve(val_vertices.size());
+    deleted_edges.reserve(val_edges.size());
+
+    std::transform(val_vertices.begin(), val_vertices.end(), std::back_inserter(deleted_vertices),
+                   [](const auto &deleted_vertex) { return VertexAccessor{deleted_vertex}; });
+    std::transform(val_edges.begin(), val_edges.end(), std::back_inserter(deleted_edges),
+                   [](const auto &deleted_edge) { return EdgeAccessor{deleted_edge}; });
```

Comment:
I dislike unpacking and repacking. 
Not essential to change ASAP, but I want to point out the idea of sending a projection into `accessor_->DetachDeleteVertexBulk` rather than unpacking first.
https://godbolt.org/z/a3foMz6ve

Reply:
I agree, however, does it make sense to add it as a separate PR on all the API methods that require such action, since it's maybe more appropriate that the API is consistent w.r.t. methods?